test/manager/upload_and_share_file_manager_test.dart

```diff
@@ -0,0 +1,166 @@
+
+    setUp(() {
+      documentRepository = MockDocumentRepository();
+      tokenRepository = MockTokenRepository();
+      credentialRepository = MockCredentialRepository();
+      uploadTaskId = UploadTaskId('upload_task_id_1');
+      uploadFileInteractor = UploadFileInteractor(
+        documentRepository,
+        tokenRepository,
+        credentialRepository,
+      );
+      shareDocumentInteractor = ShareDocumentInteractor(documentRepository);
+
+      uploadShareFileManager = UploadShareFileManager(
+        getIt.get<Store<AppState>>(),
+        uploadFileInteractor,
+        shareDocumentInteractor,
+      );
+    });
+
+    void mockPermanentUploadData() {
+      when(tokenRepository.getToken()).thenAnswer((_) async => permanentToken);
+      when(credentialRepository.getBaseUrl())
+          .thenAnswer((_) async => linShareBaseUrl);
+    }
+
+    test(
+      'justUpload just mapping result into action, then dispatch to store',
+      () async {
+        mockPermanentUploadData();
+        when(documentRepository.upload(
+          fileInfo1,
+          permanentToken,
+          linShareBaseUrl,
+        )).thenAnswer(
+          (_) async => FileUploadState(
+            Stream.fromIterable([
+              Right<Failure, Success>(fileUploadProgress10),
+              Right<Failure, Success>(fileUploadProgress100),
+              Right<Failure, Success>(FileUploadSuccess(document)),
+            ]),
+            uploadTaskId,
+          ),
+        );
+
+        await uploadShareFileManager.justUpload(fileInfo1);
+
+        expect(
+          getIt.get<Store<AppState>>().state.uploadFileState.viewState,
+          equals(Right(FileUploadSuccess(document))),
+        );
+      },
+    );
+
+    test(
+      'justUpload just mapping result into action in case failed, then dispatch to store',
+      () async {
+        final downloadFailException = Exception('Download failed exception');
+
+        mockPermanentUploadData();
+        when(documentRepository.upload(
+          fileInfo1,
+          permanentToken,
+          linShareBaseUrl,
+        )).thenAnswer(
+          (_) async => FileUploadState(
+            Stream.fromIterable([
+              Right<Failure, Success>(fileUploadProgress10),
+              Right<Failure, Success>(fileUploadProgress100),
+              Left<Failure, Success>(
+                  FileUploadFailure(fileInfo1, downloadFailException)),
+            ]),
+            uploadTaskId,
+          ),
+        );
+
+        await uploadShareFileManager.justUpload(fileInfo1);
+
+        expect(
+          getIt.get<Store<AppState>>().state.uploadFileState.viewState,
+          equals(Left(FileUploadFailure(fileInfo1, downloadFailException))),
+        );
+      },
+    );
+
+    test(
+      'justShare just mapping result into action, then dispatch to store',
+      () async {
+        when(documentRepository.share(
+          [document1.documentId],
+          [],
+          [],
```

Comment:
so weird, we can share with no one?

Reply:
IMO, we need to pre-check the `recipients` or `mailingList` before share

Reply:
You need to think about `tests` that the same with documentation. It reflects our software. Other developers will check the test first to understand what it is.